sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java

```diff
@@ -80,29 +87,42 @@ public void execute(SensorContext context) {
     } else {
       processNotebooksFiles(pythonFiles, context);
     }
+    sensorTelemetryStorage.send(context);
   }
 
   private void processNotebooksFiles(List<PythonInputFile> pythonFiles, SensorContext context) {
-    pythonFiles = parseNotebooks(pythonFiles, context);
+    pythonFiles = this.parseNotebooks(pythonFiles, context);
     // Disable caching for IPynb files for now see: SONARPY-2020
     CacheContext cacheContext = CacheContextImpl.dummyCache();
     PythonIndexer pythonIndexer = new SonarQubePythonIndexer(pythonFiles, cacheContext, context);
     PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, PythonParser.createIPythonParser(), pythonIndexer);
     scanner.execute(pythonFiles, context);
+    sensorTelemetryStorage.data.put(SensorTelemetryStorage.NOTEBOOK_RECOGNITION_ERROR_KEY, String.valueOf(scanner.getRecognitionErrorCount()));
   }
 
-  private static List<PythonInputFile> parseNotebooks(List<PythonInputFile> pythonFiles, SensorContext context) {
+  private List<PythonInputFile> parseNotebooks(List<PythonInputFile> pythonFiles, SensorContext context) {
     List<PythonInputFile> generatedIPythonFiles = new ArrayList<>();
+
+    sensorTelemetryStorage.data.put(SensorTelemetryStorage.NOTEBOOK_TOTAL_KEY, String.valueOf(pythonFiles.size()));
+    final int[] numberOfFailedParsing = {0};
```

Comment:
I guess this is a workaround as you need a final var for the lambda? I think it would be more readable to an if 
```suggestion
int numberOfFailedParsing = 0;
if(result.isPresent()){
  generatedIPythonFiles.add(result.get());
}else {
  numberOfFailedParsing++;
}
```
instead of trying to work against java.